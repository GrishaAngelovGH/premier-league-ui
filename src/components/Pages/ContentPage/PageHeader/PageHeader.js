import './PageHeader.scss'
import { Link } from 'react-router-dom'

export const ArrowLeft = () => (
    <svg width='2em' height='2em' viewBox='0 0 16 16' className='bi bi-arrow-left-short' fill='white' xmlns='http://www.w3.org/2000/svg'>
        <path fillRule='evenodd' d='M12 8a.5.5 0 0 1-.5.5H5.707l2.147 2.146a.5.5 0 0 1-.708.708l-3-3a.5.5 0 0 1 0-.708l3-3a.5.5 0 1 1 .708.708L5.707 7.5H11.5a.5.5 0 0 1 .5.5z' />
    </svg>
)

export const CloseBtn = () => (
    <span role='button'>
        <svg width='2em' height='2em' viewBox='0 0 16 16' className='bi bi-x' fill='white' xmlns='http://www.w3.org/2000/svg'>
            <path fillRule='evenodd' d='M4.646 4.646a.5.5 0 0 1 .708 0L8 7.293l2.646-2.647a.5.5 0 0 1 .708.708L8.707 8l2.647 2.646a.5.5 0 0 1-.708.708L8 8.707l-2.646 2.647a.5.5 0 0 1-.708-.708L7.293 8 4.646 5.354a.5.5 0 0 1 0-.708z' />
        </svg>
    </span>
)

const PageHeader = () => (
    <div className='row no-gutters content-header text-white justify-content-between p-3'>
        <div className='d-flex'>
            <Link to='/'>
                <ArrowLeft />
            </Link>
            <div className='m-1'>Premier League</div>
        </div>

        <CloseBtn />
    </div>
)

export default PageHeader