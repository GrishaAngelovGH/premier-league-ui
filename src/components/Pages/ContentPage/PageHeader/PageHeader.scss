.content-header {
    height: 100px;
    background: #3f1052;
}
