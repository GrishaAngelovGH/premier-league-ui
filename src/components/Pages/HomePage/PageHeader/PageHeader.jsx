import { Link } from 'react-router-dom'
import images from 'images'

const PageHeader = () => {
	const categories = ['matches', 'news', 'standings', 'stats', 'players']

	return (
		<div className='bg-purple-900 text-white h-[130px] rounded-t-md p-2'>
			<div className='flex items-center gap-2'>
				<img src={images['premierLeagueLogo']} className="w-[50px] h-[50px]" />
				<div className="text-xl">
					<div>Premier League</div>
					<div className='opacity-[0.5]'>Matchday 10 of 38</div>
				</div>
			</div>

			<div className='flex justify-evenly p-4 overflow-auto'>
				{
					categories.map((v, i) => {
						const linkClass = i === 0 ? 'border-b border-white hover:border-b-0' : 'border-b-0'

						return (
							<Link to={`/content/${v}`} key={i} className={`text-white text-decoration-none uppercase p-1 hover:bg-purple-800 hover:rounded-md ${linkClass}`}>
								{v}
							</Link>
						)
					})
				}
			</div>
		</div>
	)
}

export default PageHeader