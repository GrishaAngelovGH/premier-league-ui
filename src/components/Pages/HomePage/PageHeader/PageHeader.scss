.page-header {
    color: #ffffff;
    height: 120px;
    padding: 10px;
    background-color: #3f1052;
    border-top-left-radius: 15px;
    border-top-right-radius: 15px;

    a {
        color: #ffffff;
        &:hover {
            color: #ffffff;
            text-decoration: none;
        }
    }
}

.matchday-description {
    opacity: 0.5;
}

.item {
    width: 150px;
    height: 35px;
    text-align: center;
    text-transform: uppercase;

    &:hover {
        cursor: pointer;
        background: #390e4a;
    }
}
