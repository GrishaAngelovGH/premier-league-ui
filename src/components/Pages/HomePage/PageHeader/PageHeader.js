import { Link } from 'react-router-dom'
import images from 'images'

import './PageHeader.scss'

const PageHeader = () => {
    const categories = ['matches', 'news', 'standings', 'stats', 'players']

    return (
        <div className='page-header w-100'>
            <div className='row'>
                <div className='col-3 col-md-2 col-lg-1'>
                    <img src={images['premierLeagueLogo']} />
                </div>
                <div className='col-6 col-md-6'>
                    <div className='row'>
                        Premier League
                    </div>
                    <div className='row matchday-description'>
                        Matchday 10 of 38
                    </div>
                </div>
            </div>

            <div className='row'>
                <div className='col-md-12 justify-content-between mt-4'>
                    <div className='row flex-nowrap overflow-auto ml-1'>
                        {
                            categories.map((v, i) => {
                                const classnames = i === 0 ? 'border-bottom border-white item' : 'item'

                                return (
                                    <Link to='/content' key={i}>
                                        <div className={classnames}>{v}</div>
                                    </Link>
                                )
                            })
                        }
                    </div>
                </div>
            </div>
        </div>
    )
}

export default PageHeader