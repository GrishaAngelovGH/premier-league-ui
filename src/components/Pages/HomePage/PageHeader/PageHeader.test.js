import { render } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'

import PageHeader from './PageHeader'

test('should render component', () => {
    const container = render(<PageHeader />, { wrapper: MemoryRouter })

    expect(container).toMatchSnapshot()
})