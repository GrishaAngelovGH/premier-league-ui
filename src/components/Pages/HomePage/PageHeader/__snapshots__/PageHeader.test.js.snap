// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="page-header w-100"
      >
        <div
          class="row"
        >
          <div
            class="col-3 col-md-2 col-lg-1"
          >
            <img
              src="pl-logo.png"
            />
          </div>
          <div
            class="col-6 col-md-6"
          >
            <div
              class="row"
            >
              Premier League
            </div>
            <div
              class="row matchday-description"
            >
              Matchday 10 of 38
            </div>
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="col-md-12 justify-content-between mt-4"
          >
            <div
              class="row flex-nowrap overflow-auto ml-1"
            >
              <a
                href="/content"
              >
                <div
                  class="border-bottom border-white item"
                >
                  matches
                </div>
              </a>
              <a
                href="/content"
              >
                <div
                  class="item"
                >
                  news
                </div>
              </a>
              <a
                href="/content"
              >
                <div
                  class="item"
                >
                  standings
                </div>
              </a>
              <a
                href="/content"
              >
                <div
                  class="item"
                >
                  stats
                </div>
              </a>
              <a
                href="/content"
              >
                <div
                  class="item"
                >
                  players
                </div>
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="page-header w-100"
    >
      <div
        class="row"
      >
        <div
          class="col-3 col-md-2 col-lg-1"
        >
          <img
            src="pl-logo.png"
          />
        </div>
        <div
          class="col-6 col-md-6"
        >
          <div
            class="row"
          >
            Premier League
          </div>
          <div
            class="row matchday-description"
          >
            Matchday 10 of 38
          </div>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col-md-12 justify-content-between mt-4"
        >
          <div
            class="row flex-nowrap overflow-auto ml-1"
          >
            <a
              href="/content"
            >
              <div
                class="border-bottom border-white item"
              >
                matches
              </div>
            </a>
            <a
              href="/content"
            >
              <div
                class="item"
              >
                news
              </div>
            </a>
            <a
              href="/content"
            >
              <div
                class="item"
              >
                standings
              </div>
            </a>
            <a
              href="/content"
            >
              <div
                class="item"
              >
                stats
              </div>
            </a>
            <a
              href="/content"
            >
              <div
                class="item"
              >
                players
              </div>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
